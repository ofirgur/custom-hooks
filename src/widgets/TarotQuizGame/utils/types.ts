export interface GameStateContextInterface {
  gameState: GameState;
  setGameState: (gameState: Partial<GameState>) => void;
}

export interface GameState {
  gameScreen: GameScreen;
  gameCards: GameCard[];
  gameMatches: GameMatch[];
  gameMoves: number;
  gamePoints: number;
}

export enum GameScreen {
  game = 1,
  end = 2,
}

export interface GameCard {
  name: string;
  image: string;
  link: string;
  quizCards: GameCard[];
}

export interface GameMatch {
  gameCard: GameCard;
  matchName: string;
}

export interface NameButtonProps {
  name: string;
  disabled: boolean;
  onClick?: () => void;
}

export interface SimpleCardProps {
  image: string;
  name?: string;
  correct?: boolean;
}
