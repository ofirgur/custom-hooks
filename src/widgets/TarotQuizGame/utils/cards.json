{
  "cards": [
    {
      "name": "The Fool",
      "image": "https://storage.googleapis.com/wz_email_media/images/the_fool_quiz.png"
    },
    {
      "name": "The Magician",
      "image": "https://storage.googleapis.com/wz_email_media/images/the_magician_quiz.png"
    },
    {
      "name": "The High Priestess",
      "image": "https://storage.googleapis.com/wz_email_media/images/the_high_priestess_quiz.png"
    },
    {
      "name": "The Empress",
      "image": "https://storage.googleapis.com/wz_email_media/images/the_empress_quiz.png"
    },
    {
      "name": "The Emperor",
      "image": "https://storage.googleapis.com/wz_email_media/images/the_emperor_quiz.png"
    },
    {
      "name": "The Hierophant",
      "image": "https://storage.googleapis.com/wz_email_media/images/the_hierophant_quiz.png"
    },
    {
      "name": "The Lovers",
      "image": "https://storage.googleapis.com/wz_email_media/images/the_lovers_quiz.png"
    },
    {
      "name": "The Chariot",
      "image": "https://storage.googleapis.com/wz_email_media/images/the_chariot_quiz.png"
    },
    {
      "name": "Strength",
      "image": "https://storage.googleapis.com/wz_email_media/images/strength_quiz.png"
    },
    {
      "name": "The Hermit",
      "image": "https://storage.googleapis.com/wz_email_media/images/the_hermit_quiz.png"
    },
    {
      "name": "Wheel of Fortune",
      "image": "https://storage.googleapis.com/wz_email_media/images/wheel_of_fortune_quiz.png"
    },
    {
      "name": "Justice",
      "image": "https://storage.googleapis.com/wz_email_media/images/justice_quiz.png"
    },
    {
      "name": "The Hanged Man",
      "image": "https://storage.googleapis.com/wz_email_media/images/the_hanged_man_quiz.png"
    },
    {
      "name": "Death",
      "image": "https://storage.googleapis.com/wz_email_media/images/death_quiz.png"
    },
    {
      "name": "Temperance",
      "image": "https://storage.googleapis.com/wz_email_media/images/temperance_quiz.png"
    },
    {
      "name": "The Devil",
      "image": "https://storage.googleapis.com/wz_email_media/images/the_devil_quiz.png"
    },
    {
      "name": "The Tower",
      "image": "https://storage.googleapis.com/wz_email_media/images/the_tower_quiz.png"
    },
    {
      "name": "The Star",
      "image": "https://storage.googleapis.com/wz_email_media/images/the_star_quiz.png"
    },
    {
      "name": "The Moon",
      "image": "https://storage.googleapis.com/wz_email_media/images/the_moon_quiz.png"
    },
    {
      "name": "The Sun",
      "image": "https://storage.googleapis.com/wz_email_media/images/the_sun_quiz.png"
    },
    {
      "name": "Judgement",
      "image": "https://storage.googleapis.com/wz_email_media/images/judgement_quiz.png"
    },
    {
      "name": "The World",
      "image": "https://storage.googleapis.com/wz_email_media/images/the_world_quiz.png"
    },
    {
      "name": "Ace of Wands",
      "image": "https://storage.googleapis.com/wz_email_media/images/ace_of_wands_quiz.png"
    },
    {
      "name": "Two of Wands",
      "image": "https://storage.googleapis.com/wz_email_media/images/two_of_wands_quiz.png"
    },
    {
      "name": "Three of Wands",
      "image": "https://storage.googleapis.com/wz_email_media/images/three_of_wands_quiz.png"
    },
    {
      "name": "Fourth of Wands",
      "image": "https://storage.googleapis.com/wz_email_media/images/four_of_wands_quiz.png"
    },
    {
      "name": "Five of Wands",
      "image": "https://storage.googleapis.com/wz_email_media/images/five_of_wands_quiz.png"
    },
    {
      "name": "Six of Wands",
      "image": "https://storage.googleapis.com/wz_email_media/images/six_of_wands_quiz.png"
    },
    {
      "name": "Seven of Wands",
      "image": "https://storage.googleapis.com/wz_email_media/images/seven_of_wands_quiz.png"
    },
    {
      "name": "Eight of Wands",
      "image": "https://storage.googleapis.com/wz_email_media/images/eight_of_wands_quiz.png"
    },
    {
      "name": "Nine of Wands",
      "image": "https://storage.googleapis.com/wz_email_media/images/nine_of_wands_quiz.png"
    },
    {
      "name": "Ten of Wands",
      "image": "https://storage.googleapis.com/wz_email_media/images/ten_of_wands_quiz.png"
    },
    {
      "name": "Page of Wands",
      "image": "https://storage.googleapis.com/wz_email_media/images/page_of_wands_quiz.png"
    },
    {
      "name": "Knight of Wands",
      "image": "https://storage.googleapis.com/wz_email_media/images/knight_of_wands_quiz.png"
    },
    {
      "name": "Queen of Wands",
      "image": "https://storage.googleapis.com/wz_email_media/images/queen_of_wands_quiz.png"
    },
    {
      "name": "King of Wands",
      "image": "https://storage.googleapis.com/wz_email_media/images/king_of_wands_quiz.png"
    },
    {
      "name": "Ace of Cups",
      "image": "https://storage.googleapis.com/wz_email_media/images/ace_of_cups_quiz.png"
    },
    {
      "name": "Two of Cups",
      "image": "https://storage.googleapis.com/wz_email_media/images/two_of_cups_quiz.png"
    },
    {
      "name": "Three of Cups",
      "image": "https://storage.googleapis.com/wz_email_media/images/three_of_cups_quiz.png"
    },
    {
      "name": "Four of Cups",
      "image": "https://storage.googleapis.com/wz_email_media/images/four_of_cups_quiz.png"
    },
    {
      "name": "Five of Cups",
      "image": "https://storage.googleapis.com/wz_email_media/images/five_of_cups_quiz.png"
    },
    {
      "name": "Six of Cups",
      "image": "https://storage.googleapis.com/wz_email_media/images/six_of_cups_quiz.png"
    },
    {
      "name": "Seven of Cups",
      "image": "https://storage.googleapis.com/wz_email_media/images/seven_of_cups_quiz.png"
    },
    {
      "name": "Eight of Cups",
      "image": "https://storage.googleapis.com/wz_email_media/images/eight_of_cups_quiz.png"
    },
    {
      "name": "Nine of Cup",
      "image": "https://storage.googleapis.com/wz_email_media/images/nine_of_cups_quiz.png"
    },
    {
      "name": "Ten of Cups",
      "image": "https://storage.googleapis.com/wz_email_media/images/ten_of_cups_quiz.png"
    },
    {
      "name": "Page of Cups",
      "image": "https://storage.googleapis.com/wz_email_media/images/page_of_cups_quiz.png"
    },
    {
      "name": "Knight of Cups",
      "image": "https://storage.googleapis.com/wz_email_media/images/knight_of_cups_quiz.png"
    },
    {
      "name": "Queen of Cups",
      "image": "https://storage.googleapis.com/wz_email_media/images/queen_of_cups_quiz.png"
    },
    {
      "name": "King of Cups",
      "image": "https://storage.googleapis.com/wz_email_media/images/king_of_cups_quiz.png"
    },
    {
      "name": "Ace of Swords",
      "image": "https://storage.googleapis.com/wz_email_media/images/ace_of_swords_quiz.png"
    },
    {
      "name": "Two of Swords",
      "image": "https://storage.googleapis.com/wz_email_media/images/two_of_swords_quiz.png"
    },
    {
      "name": "Three of Swords",
      "image": "https://storage.googleapis.com/wz_email_media/images/three_of_swords_quiz.png"
    },
    {
      "name": "Four of Swords",
      "image": "https://storage.googleapis.com/wz_email_media/images/four_of_swords_quiz.png"
    },
    {
      "name": "Five of Swords",
      "image": "https://storage.googleapis.com/wz_email_media/images/five_of_swords_quiz.png"
    },
    {
      "name": "Six of Swords",
      "image": "https://storage.googleapis.com/wz_email_media/images/six_of_swords_quiz.png"
    },
    {
      "name": "Seven of Swords",
      "image": "https://storage.googleapis.com/wz_email_media/images/seven_of_swords_quiz.png"
    },
    {
      "name": "Eight of Swords",
      "image": "https://storage.googleapis.com/wz_email_media/images/eight_of_swords_quiz.png"
    },
    {
      "name": "Nine of Swords",
      "image": "https://storage.googleapis.com/wz_email_media/images/nine_of_swords_quiz.png"
    },
    {
      "name": "Ten of Swords",
      "image": "https://storage.googleapis.com/wz_email_media/images/ten_of_swords_quiz.png"
    },
    {
      "name": "Page of Swords",
      "image": "https://storage.googleapis.com/wz_email_media/images/page_of_swords_quiz.png"
    },
    {
      "name": "Knight of Swords",
      "image": "https://storage.googleapis.com/wz_email_media/images/knight_of_swords_quiz.png"
    },
    {
      "name": "Queen of Swords",
      "image": "https://storage.googleapis.com/wz_email_media/images/queen_of_swords_quiz.png"
    },
    {
      "name": "King of Swords",
      "image": "https://storage.googleapis.com/wz_email_media/images/king_of_swords_quiz.png"
    },
    {
      "name": "Ace of Pentacles",
      "image": "https://storage.googleapis.com/wz_email_media/images/ace_of_pentacles_quiz.png"
    },
    {
      "name": "Two of Pentacles",
      "image": "https://storage.googleapis.com/wz_email_media/images/two_of_pentacles_quiz.png"
    },
    {
      "name": "Three of Pentacles",
      "image": "https://storage.googleapis.com/wz_email_media/images/three_of_pentacles_quiz.png"
    },
    {
      "name": "Four of Pentacles",
      "image": "https://storage.googleapis.com/wz_email_media/images/four_of_pentacles_quiz.png"
    },
    {
      "name": "Five of Pentacles",
      "image": "https://storage.googleapis.com/wz_email_media/images/five_of_pentacles_quiz.png"
    },
    {
      "name": "Six of Pentacles",
      "image": "https://storage.googleapis.com/wz_email_media/images/six_of_pentacles_quiz.png"
    },
    {
      "name": "Seven of Pentacles",
      "image": "https://storage.googleapis.com/wz_email_media/images/seven_of_pentacles_quiz.png"
    },
    {
      "name": "Eight of Pentacles",
      "image": "https://storage.googleapis.com/wz_email_media/images/eight_of_pentacles_quiz.png"
    },
    {
      "name": "Nine of Pentacles",
      "image": "https://storage.googleapis.com/wz_email_media/images/nine_of_pentacles_quiz.png"
    },
    {
      "name": "Ten of Pentacles",
      "image": "https://storage.googleapis.com/wz_email_media/images/ten_of_pentacles_quiz.png"
    },
    {
      "name": "Page of Pentacles",
      "image": "https://storage.googleapis.com/wz_email_media/images/page_of_pentacles_quiz.png"
    },
    {
      "name": "Knight of Pentacles",
      "image": "https://storage.googleapis.com/wz_email_media/images/knight_of_pentacles_quiz.png"
    },
    {
      "name": "Queen of Pentacles",
      "image": "https://storage.googleapis.com/wz_email_media/images/queen_of_pentacles_quiz.png"
    },
    {
      "name": "King of Pentacles",
      "image": "https://storage.googleapis.com/wz_email_media/images/king_of_pentacles_quiz.png"
    }
  ]
}
