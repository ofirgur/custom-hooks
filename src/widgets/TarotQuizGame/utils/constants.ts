import { GameState, GameScreen } from './types';

export const INITIAL_GAME_STATE: GameState = {
  gameScreen: GameScreen.game,
  gameCards: [],
  gameMatches: [],
  gameMoves: 0,
  gamePoints: 0,
};
export const CARDS_NUMBER = 6;
export const QUIZ_CARDS_NUMBER = 3;
export const QUIZ_INDICATOR_ARRAY = [0, 1, 2, 3, 4, 5];
export const WAZIMO_ASSETS_URL = 'https://cdn.mmctsvc.com/email/tarot-memory-game.json';
export const YES_NO_TAROT_URL = 'https://all-tarot.com/readings/yes-or-no-tarot/';
export const GAME_LAST_MATCHE = QUIZ_INDICATOR_ARRAY.length - 1;

export const GAME_POINTS = {
  AMAIZING: 1200,
  WELL_DONE: 1000,
  WELL_DONE_2: 800,
  NOT_BAD: 600,
  NOT_BAD_2: 400,
  TRY_AGAIN: 200,
  TRY_AGAIN_2: 0,
};

export const GAME_STATE_LOCAL_STORAGE_NAME = 'tarrot-quiz-game-state';

export const SCORE_PAGE_URL = window.location.origin.concat(window.location.pathname).concat('/score').concat(window.location.search);
export const GAME_PAGE_URL = window.location.origin.concat(window.location.pathname).replace('/score', '');
