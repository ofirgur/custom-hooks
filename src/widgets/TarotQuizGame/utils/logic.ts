import { GameCard, GameMatch } from './types';

export const shuffleArray = (array: any[]) => {
  return array
    .map((a: any) => ({ sort: Math.random(), value: a }))
    .sort((a: any, b: any) => a.sort - b.sort)
    .map((a: any) => a.value);
};

const getRandom = (n: number, array: any) => {
  return array
    .sort(() => Math.random() - Math.random())
    .slice(0, n);
};

export const getRandomGameCards = (numberOfCards: number, numberOfQuizCards: number, cards: GameCard[]) => {
  return getRandom(numberOfCards, cards)
    .map((card: GameCard) => ({
      ...card,
      quizCards: shuffleArray([
        card,
        ...getRandom(numberOfQuizCards, cards.filter(c => c.name !== card.name)),
      ]),
    }));
};

export const getGamePoints = (matches: GameMatch[]) => {
  return matches
    .filter((match: GameMatch) => match.gameCard.name === match.matchName)
    .length * 200;
};
