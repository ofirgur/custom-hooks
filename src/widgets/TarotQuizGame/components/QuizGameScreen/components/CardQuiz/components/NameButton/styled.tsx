import styled from '@emotion/styled';

import { VARIABLES } from '../../../../../../utils/variables';

type NameButtonStyledProps = {
  disabled: boolean;
};

export const NameButtonStyled = styled.button<NameButtonStyledProps>`
  padding: 5px;
  width: 200px;
  max-width: 300px;
  height: 54px;
  background: ${VARIABLES.BACKGROUND.BLACK.BUTTON};
  color: ${VARIABLES.COLOR.WHITE};
  box-shadow: ${VARIABLES.BOX_SHADOW.BUTTON};
  border: 0;
  border-radius: 6px;
  text-align: center;
  font-size: 18px;
  line-height: 22px;
  font-weight: 400;
  visibility: ${props => (props.disabled ? 'hidden' : 'visible')};
  cursor: pointer;
`;
