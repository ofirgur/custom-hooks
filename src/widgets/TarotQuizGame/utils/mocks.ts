
import { GameScreen } from './types';
import { GAME_POINTS } from './constants';

export const mockedGameState = {
  gameScreen: GameScreen.game,
  gameMatches: [],
  gameMoves: 0,
  gamePoints: 0,
  gameCards: [],
};

export const mockedNumbersArray = [0, 1, 2, 3, 4, 5];
export const mockedCardIndex = 2;
export const mockedPoints = GAME_POINTS.AMAIZING;

export const mockedGameCard = {
  "name": "The Fool",
  "image": "https://storage.googleapis.com/wz_email_media/images/the_fool_quiz.png",
  quizCards: [
    {
      "name": "The Fool",
      "image": "https://storage.googleapis.com/wz_email_media/images/the_fool_quiz.png",
    },
    {
      "name": "The Magician",
      "image": "https://storage.googleapis.com/wz_email_media/images/the_magician_quiz.png"
    },
    {
      "name": "The High Priestess",
      "image": "https://storage.googleapis.com/wz_email_media/images/the_high_priestess_quiz.png"
    },
    {
      "name": "The Empress",
      "image": "https://storage.googleapis.com/wz_email_media/images/the_empress_quiz.png"
    }
  ],
};

export const mockedGameCards = [
  {
    "name": "The Fool",
    "image": "https://storage.googleapis.com/wz_email_media/images/the_fool_quiz.png",
    quizCards: [
      {
        "name": "The Fool",
        "image": "https://storage.googleapis.com/wz_email_media/images/the_fool_quiz.png",
      },
      {
        "name": "The Magician",
        "image": "https://storage.googleapis.com/wz_email_media/images/the_magician_quiz.png"
      },
      {
        "name": "The High Priestess",
        "image": "https://storage.googleapis.com/wz_email_media/images/the_high_priestess_quiz.png"
      },
      {
        "name": "The Empress",
        "image": "https://storage.googleapis.com/wz_email_media/images/the_empress_quiz.png"
      }
    ],
  },
  {
    "name": "The Magician",
    "image": "https://storage.googleapis.com/wz_email_media/images/the_magician_quiz.png",
    quizCards: [
      {
        "name": "The Fool",
        "image": "https://storage.googleapis.com/wz_email_media/images/the_fool_quiz.png",
      },
      {
        "name": "The Magician",
        "image": "https://storage.googleapis.com/wz_email_media/images/the_magician_quiz.png"
      },
      {
        "name": "The High Priestess",
        "image": "https://storage.googleapis.com/wz_email_media/images/the_high_priestess_quiz.png"
      },
      {
        "name": "The Empress",
        "image": "https://storage.googleapis.com/wz_email_media/images/the_empress_quiz.png"
      }
    ],
  },
  {
    "name": "The High Priestess",
    "image": "https://storage.googleapis.com/wz_email_media/images/the_high_priestess_quiz.png",
    quizCards: [
      {
        "name": "The Fool",
        "image": "https://storage.googleapis.com/wz_email_media/images/the_fool_quiz.png",
      },
      {
        "name": "The Magician",
        "image": "https://storage.googleapis.com/wz_email_media/images/the_magician_quiz.png"
      },
      {
        "name": "The High Priestess",
        "image": "https://storage.googleapis.com/wz_email_media/images/the_high_priestess_quiz.png"
      },
      {
        "name": "The Empress",
        "image": "https://storage.googleapis.com/wz_email_media/images/the_empress_quiz.png"
      }
    ],
  },
  {
    "name": "The Empress",
    "image": "https://storage.googleapis.com/wz_email_media/images/the_empress_quiz.png",
    quizCards: [
      {
        "name": "The Fool",
        "image": "https://storage.googleapis.com/wz_email_media/images/the_fool_quiz.png",
      },
      {
        "name": "The Magician",
        "image": "https://storage.googleapis.com/wz_email_media/images/the_magician_quiz.png"
      },
      {
        "name": "The High Priestess",
        "image": "https://storage.googleapis.com/wz_email_media/images/the_high_priestess_quiz.png"
      },
      {
        "name": "The Empress",
        "image": "https://storage.googleapis.com/wz_email_media/images/the_empress_quiz.png"
      }
    ],
  },
  {
    "name": "The Empress",
    "image": "https://storage.googleapis.com/wz_email_media/images/the_empress_quiz.png",
    quizCards: [
      {
        "name": "The Fool",
        "image": "https://storage.googleapis.com/wz_email_media/images/the_fool_quiz.png",
      },
      {
        "name": "The Magician",
        "image": "https://storage.googleapis.com/wz_email_media/images/the_magician_quiz.png"
      },
      {
        "name": "The High Priestess",
        "image": "https://storage.googleapis.com/wz_email_media/images/the_high_priestess_quiz.png"
      },
      {
        "name": "The Empress",
        "image": "https://storage.googleapis.com/wz_email_media/images/the_empress_quiz.png"
      }
    ],
  }, {
    "name": "The Empress",
    "image": "https://storage.googleapis.com/wz_email_media/images/the_empress_quiz.png",
    quizCards: [
      {
        "name": "The Fool",
        "image": "https://storage.googleapis.com/wz_email_media/images/the_fool_quiz.png",
      },
      {
        "name": "The Magician",
        "image": "https://storage.googleapis.com/wz_email_media/images/the_magician_quiz.png"
      },
      {
        "name": "The High Priestess",
        "image": "https://storage.googleapis.com/wz_email_media/images/the_high_priestess_quiz.png"
      },
      {
        "name": "The Empress",
        "image": "https://storage.googleapis.com/wz_email_media/images/the_empress_quiz.png"
      }
    ],
  },
];

export const mockedMatches = [
  {
    gameCard: {
      "name": "The Fool",
      "image": "https://storage.googleapis.com/wz_email_media/images/the_fool_quiz.png",
    },
    matchName: "The Fool",
  },
  {
    gameCard: {
      "name": "The Magician",
      "image": "https://storage.googleapis.com/wz_email_media/images/the_fool_quiz.png",
    },
    matchName: "The Fool",
  },
  {
    gameCard: {
      "name": "The Fool",
      "image": "https://storage.googleapis.com/wz_email_media/images/the_fool_quiz.png",
    },
    matchName: "The Fool",
  },
  {
    gameCard: {
      "name": "The Fool",
      "image": "https://storage.googleapis.com/wz_email_media/images/the_fool_quiz.png",
    },
    matchName: "The Fool",
  },
  {
    gameCard: {
      "name": "The Fool",
      "image": "https://storage.googleapis.com/wz_email_media/images/the_fool_quiz.png",
    },
    matchName: "The Fool",
  },
  {
    gameCard: {
      "name": "The Fool",
      "image": "https://storage.googleapis.com/wz_email_media/images/the_fool_quiz.png",
    },
    matchName: "The Fool",
  },
];
