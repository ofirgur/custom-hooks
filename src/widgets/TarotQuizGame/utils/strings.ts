import { GAME_POINTS } from './constants';

export const STRINGS = {
  GAME_SCREEN: {
    MATCHES: 'Matches',
    MOVES: 'Moves',
    FETCHING: 'Fetching...',
    NEXT: 'Next',
    GET_YOUR_SCORE: 'Get Your Score',
  },
  END_SCREEN: {
    START_OVER: 'Start Over',
    YES_NO_TAROT: 'Yes No Tarot',
    READ_MORE: 'Read more about the cards you were able to match',
    MOVES: 'Moves',
    POINTS: 'Points',
    STRING_PER_SCORE: (score: number) => {
      switch (score) {
        case GAME_POINTS.TRY_AGAIN_2: return 'Better luck next time! You got 0/6 answers correctï»¿';
        case GAME_POINTS.TRY_AGAIN: return 'Better luck next time! You got 1/6 answers correct';
        case GAME_POINTS.NOT_BAD_2: return "You're getting there! You got 2/6 answers correct";
        case GAME_POINTS.NOT_BAD: return "You're getting there! You got 3/6 answers correct";
        case GAME_POINTS.WELL_DONE_2: return 'Look at you go! You got 4/6 answers correct';
        case GAME_POINTS.WELL_DONE: return 'Look at you go! You got 5/6 answers correct';
        case GAME_POINTS.AMAIZING: return "Wow! You're a tarot expert. 6/6 answers correct!";
        default: return '';
      }
    },
    EVENT_PER_SCORE: (score: number) => {
      switch (score) {
        case GAME_POINTS.TRY_AGAIN_2: return 'try-again';
        case GAME_POINTS.TRY_AGAIN: return 'try-again';
        case GAME_POINTS.NOT_BAD_2: return 'not-bad';
        case GAME_POINTS.NOT_BAD: return 'not-bad';
        case GAME_POINTS.WELL_DONE_2: return 'well-done';
        case GAME_POINTS.WELL_DONE: return 'well-done';
        case GAME_POINTS.AMAIZING: return 'amazing';
        default: return '';
      }
    },
  },
};
