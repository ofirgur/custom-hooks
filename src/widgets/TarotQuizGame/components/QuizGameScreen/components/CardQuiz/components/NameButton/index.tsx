import React from 'react';
import { NameButtonProps } from '../../../../../../utils/types';
import { NameButtonStyled } from './styled';

const NameButton = (props: NameButtonProps) => {
  const { name, onClick, disabled } = props;
  const handleClick = () => {
    onClick && onClick();
  };

  return (
    <NameButtonStyled
      onClick={handleClick}
      disabled={disabled}
    >
      {name}
    </NameButtonStyled>
  );
};

export default NameButton;
