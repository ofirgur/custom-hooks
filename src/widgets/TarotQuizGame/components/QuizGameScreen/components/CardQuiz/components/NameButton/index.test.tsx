import * as React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { mockedGameCard } from '../../../../../../utils/mocks';
import NameButton from './';

const mockedOnClick = jest.fn();
const props = {
  name: mockedGameCard.name,
  onClick: mockedOnClick,
  disabled: false,
};

describe('NameButton', () => {
  it('should render correctly', () => {
    const { container } = render(<NameButton {...props} />);
    expect(container).toMatchSnapshot();
  });

  it('should call onClick on user click', async () => {
    const { getByText } = render(<NameButton {...props} />);
    const user = userEvent.setup();
    await user.click(getByText(mockedGameCard.name));
    expect(mockedOnClick).toHaveBeenCalled();
  });
});
