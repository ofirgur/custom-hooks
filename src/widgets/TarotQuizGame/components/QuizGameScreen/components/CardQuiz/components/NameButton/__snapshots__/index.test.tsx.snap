// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameButton should render correctly 1`] = `
<div>
  <button
    class="css-1o4p7rv"
  >
    The Fool
  </button>
</div>
`;
